pkg/controllers/jobset_controller.go

```diff
@@ -484,30 +497,20 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 		// This updates the StartupPolicy condition and notifies that we are waiting
 		// for this replicated job to finish.
 		if !jobSetSuspended(js) && inOrderStartupPolicy(startupPolicy) {
-			return r.ensureCondition(ctx, ensureConditionOpts{
-				jobset:           js,
-				eventType:        corev1.EventTypeNormal,
-				forceFalseUpdate: true,
-				condition:        inOrderStartupPolicyExecutingCondition(),
-			})
+			setInOrderStartupPolicyInProgress(js, updateStatusOpts)
+			return nil
 		}
 	}
 	allErrs := errors.Join(finalErrs...)
 	if allErrs != nil {
-		// Emit event to propagate the Job creation failures up to be more visible to the user.
```

Comment:
Note for reveiwers: I removed this event since the one we added in #466 (name of first failed job that led to jobset failure) is more useful than this (and this event potentially got very large as well since it contained all errors from parallel job creations).